app/scripts/modules/core/src/cluster/ClusterPostSearchResultSearcher.ts

```diff
@@ -42,14 +48,51 @@ export class ClusterPostSearchResultSearcher implements IPostSearchResultSearche
       }
     }), 'cluster');
 
-    const formatter: ISearchResultFormatter = searchResultFormatterRegistry.get(ClusterPostSearchResultSearcher.TYPE);
-    return this.$q.when([{
-      id: ClusterPostSearchResultSearcher.TYPE,
-      category: ClusterPostSearchResultSearcher.TYPE,
-      icon: formatter.icon,
-      iconClass: '',
-      order: formatter.order,
-      results: clusters
-    }]);
+    const requests: IPromise<IServerGroup>[] =
+      inputs.map((serverGroup: IServerGroupSearchResult) => {
+        return this.serverGroupReader.getServerGroup(serverGroup.application, serverGroup.account, serverGroup.region, serverGroup.serverGroup)
+          .then((sg: IServerGroup) => {
+            serverGroup.exists = true;
+            return sg;
+          })
+          .catch(() => {
+            serverGroup.exists = false;
+            return null;
+          });
+      });
+    return this.$q.all(requests)
+      .then((serverGroups: IServerGroup[]) => {
```

Comment:
Creates a multimap by cluster, adding all the existing ASGs to its corresponding cluster.